plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TrinoHiveCatalog.java

```diff
@@ -298,13 +303,51 @@ public Transaction newCreateTableTransaction(
     public void dropTable(ConnectorSession session, SchemaTableName schemaTableName, boolean purgeData)
     {
         // TODO: support path override in Iceberg table creation: https://github.com/trinodb/trino/issues/8861
-        Table table = loadTable(session, schemaTableName);
+        TableMetadata metadata = tableMetadataCache.computeIfAbsent(
+                schemaTableName,
+                ignore -> ((BaseTable) loadIcebergTable(metastore, tableOperationsProvider, session, schemaTableName)).operations().current());
+
+        Table table = getIcebergTableWithMetadata(metastore, tableOperationsProvider, session, schemaTableName, metadata);
         if (table.properties().containsKey(OBJECT_STORE_PATH) ||
                 table.properties().containsKey(WRITE_NEW_DATA_LOCATION) ||
                 table.properties().containsKey(WRITE_METADATA_LOCATION)) {
             throw new TrinoException(NOT_SUPPORTED, "Table " + schemaTableName + " contains Iceberg path override properties and cannot be dropped from Trino");
         }
-        metastore.dropTable(schemaTableName.getSchemaName(), schemaTableName.getTableName(), purgeData);
+        io.trino.plugin.hive.metastore.Table metastoreTable = metastore.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName())
+                .orElseThrow(() -> new TableNotFoundException(schemaTableName));
+
+        metastore.dropTable(
+                schemaTableName.getSchemaName(),
+                schemaTableName.getTableName(),
+                false /* do not delete data */);
+        if (purgeData) {
```

Comment:
`purgeData` is always true (https://github.com/trinodb/trino/pull/11167), so:

```suggestion
        verify(purgeData); // purgeData is always true
```